source/leuven/services/openondemand.rst

```diff
@@ -191,136 +220,252 @@ of using any of the other partitions. There are however some limitations on the
+In the app resource form, besides the normal choices (:ref:`listed above <interactive-apps>`), you can also choose
+a 'Toolchain year' such as '2023a' from a drop-down menu.
+Based on that choice, a correct JupyterLab module will be loaded together with its dependencies (specifically Python).
+To reproduce your numerical results at any time in the future, make sure you stick to choosing the same 'Toolchain year'.
+
+Furthermore, you may optionally choose to load ``SciPy-bundle`` (for widely-used packages like ``scipy``,
+``numpy``, ``pandas`` and more) and/or ``matplotlib`` in your environment from the same 'Toolchain year'.
+
+Once you launch a JupyterLab session, a default kernel called ``Python 3 (ipykernel)`` is already available in your session.
+This kernel, in addition to the Python standard library, would enable using extra packages from
+``SciPy-bundle`` and/or ``matplotlib``, if you already selected them.
+
+If the "standard" environment explained above does not provide all packages that you need, then
+it is recommended to manage your custom-made R or Python environments using Conda.
+You should :ref:`install Miniconda <install_miniconda_python>` if you have not installed it yet.
 
-To create any other kernels, first create a  :ref:`Python <create_python_conda_env>` or 
-:ref:`R <create_r_conda_env>` conda environment. The second step consists out of effectively
-creating the kernel in your ``$VSC_HOME/.local`` folder, as Jupyter will look for kernels in this location. The following commands should be excecuted from a shell, 
-and only need to be done once for the set-up of each new kernel. This starts with activating your conda
-environment::
+To create any other kernel, first create a  :ref:`Python <create_python_conda_env>` or
+:ref:`R <create_r_conda_env>` Conda environment. The second step consists out of effectively
+creating the kernel in your ``$VSC_HOME/.local`` folder, as Jupyter will look for kernels in this location.
+The following commands should be excecuted from a shell (e.g. using 'Login Server Shell Access'), and only need
+to be done once for the set-up of each new kernel.
+If you already have an existing Python kernel, but your JupyterLab session freeses/craches when choosing your
+old kernel, you also need to repeat the following steps only once.
+This starts with activating your Conda environment::
 
       source activate <env_name>
 
-For Python you will need the ``ipykernel`` package installed in your conda environment. Then you create the kernel as follows::
+For Python you will need the ``ipykernel`` package installed in your Conda environment::
 
-      python -m ipykernel install --user --name '<env_name>' --display-name '<kernel_name>'
+      conda install ipykernel
+
+Then you create the kernel as follows::
+
+      python -m ipykernel install --user --env PYTHONPATH "" --name '<env_name>' --display-name '<kernel_name>'
 
```

Comment:
I tried with an older kernel (Python 3.6.8, ipython 7.6.1,ipykernel 5.1.1) and got an error: 
 ipython-kernel-install: error: unrecognized arguments: --env PYTHONPATH
with a newer kernel (Python 3.8.16, ipython 8.8.0, ipykernel 6.19.2) re-registering worked and solved the issue. 

do we need  a minimum version to fix this?

Reply:
perhaps, yes.

Reply:
I checked few conda environments, and I updated the text accordingly:

"The minimum supported version for Python is 3.7 and for ``ipykernel`` package is 6.19.2."

Reply:
To make it very clear put it in a "note: " ?